File: ios/PetitesGouttes/ViewModels/SettingsViewModel.swift
import SwiftUI

@Observable
@MainActor
final class SettingsViewModel {
    @ObservationIgnored
    @AppStorage("lowStockThresholdMl") var lowStockThresholdMl: Int = 1500

    @ObservationIgnored
    @AppStorage("dailyConsumptionMl") var dailyConsumptionMl: Int = 300

    @ObservationIgnored
    @AppStorage("daycareDaysPerWeek") var daycareDaysPerWeek: Int = 5
}

File: app/src/main/java/com/bnjdpn/petitesgouttes/viewmodel/SettingsViewModel.kt
package com.bnjdpn.petitesgouttes.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.viewModelScope
import com.bnjdpn.petitesgouttes.data.preferences.SettingsDataStore
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.launch

class SettingsViewModel(application: Application) : AndroidViewModel(application) {
    private val settings = SettingsDataStore(application)

    val lowStockThreshold: Flow<Int> = settings.lowStockThresholdMl
    val dailyConsumption: Flow<Int> = settings.dailyConsumptionMl
    val daycareDays: Flow<Int> = settings.daycareDaysPerWeek

    fun updateLowStockThreshold(value: Int) {
        viewModelScope.launch {
            settings.setLowStockThreshold(value)
        }
    }

    fun updateDailyConsumption(value: Int) {
        viewModelScope.launch {
            settings.setDailyConsumption(value)
        }
    }

    fun updateDaycareDays(value: Int) {
        viewModelScope.launch {
            settings.setDaycareDaysPerWeek(value)
        }
    }
}
